### ISSUE TITLE
onError reports with wrong type

### ISSUE THREAD

**Issue by suterma** [vue-youtube/vue-youtube#9]
Hi, great library. I have small issue when handling the onError reponses. I want to get the event.data value, then handle it as the PlayerError type.

However, when receiving the data at runtime, the error's data property actually contains a string, not the PlayerError enum value.

I suggest, you make an addional parsing step, to make sure, that the data with the correct type is returned.

**Comment by suterma** [comment]
In the meantime, I have written myself this workaround, FYI, or anyone interested:

```
/** Handle YouTube IFrame player errors
 * See https://developers.google.com/youtube/iframe_api_reference#Events for details
 */
onError((event) => {
    const url = instance.value?.getVideoUrl();
    const errorCode = PlayerError[event.data];
    switch (errorCode) {
        case PlayerError[PlayerError.INVALID_PARAMETER]:
            console.error(
                `The request contains an invalid parameter value. Is the given YouTube URL '${url}' working on youtube.com?`,
            );
            break;
        case PlayerError[PlayerError.HTML5_ERROR]:
            console.error(
                `The requested content from YouTube URL '${url}' cannot be played in Replayer. Try another video.`,
            );
            break;
        case PlayerError[PlayerError.NOT_FOUND]:
            console.error(
                `The requested video from YouTube URL '${url}' was not found. This error occurs when a video has been removed (for any reason) or has been marked as private. Try another video.`,
            );
            break;
        case PlayerError[PlayerError.NOT_ALLOWED]:
        case PlayerError[PlayerError.NOT_ALLOWED_DISGUISE]:
        case '150':
            console.error(
                `The owner of the requested video from YouTube URL '${url}' does not allow it to be played in embedded players like Replayer. Try another video or watch it online on YouTube (without Replayer).`,
            );
            break;
        default:
            console.error('TrackYoutubeElement::onStateChange:onError:', event);
    }
});
```

**Comment by Techassi** [comment]
Glad to hear you enjoy using my package :)

I can look into improving the error handling/parsing om my side.

**Comment by Techassi** [comment]
I took a look, and you can directly match the enum types using TypeScript. The returned error value during runtime is a number.

```vue
<template>
  <div ref="yt" />
</template>

<script setup lang="ts">
import { PlayerError, usePlayer } from '@vue-youtube/core';
import { ref } from 'vue';

const videoId = ref('dQw4w9WgXcQ');
const yt = ref();

const { onError } = usePlayer(videoId, yt, {
  playerVars: {
    autoplay: 1,
    mute: 1,
  },
});

setTimeout(() => {
  videoId.value = 'lol';
}, 5000);

onError((error) => {
  switch (error.data) {
    case PlayerError.INVALID_PARAMETER: {
      break;
    }
    case PlayerError.HTML5_ERROR: {
      break;
    }
    case PlayerError.NOT_FOUND:{
      break;
    }
    case PlayerError.NOT_ALLOWED:{
      break;
    }
    case PlayerError.NOT_ALLOWED_DISGUISE: {
      break;
    }
  }
});
</script>
```

Let me know if you need any further help!

**Comment by Techassi** [comment]
Can this be closed @suterma? Could you make it work with my suggested code snippet?

**Comment by suterma** [comment]
Unfortunately, this did not solve the problem for me. I still receive error.data as string type. See the attached screenshot.
![Screenshot from 2023-08-28 21-52-20](https://github.com/vue-youtube/vue-youtube/assets/11473920/4138a034-ea73-4484-bd3f-6d1edce9237f)
I am debugging on a updated google chrome browser on Ubuntu. The data property type at runtime is actually a string, and It only works for me when I cast the error enum to a string beforehand. This is still the version 0.0.4 of the package.

Very strange. Maybe I am the only one with this problem. If not, the workaround can be found above. Feel free to close the issue.

**Comment by Techassi** [comment]
Could you try the above snippet in a fresh Vue 3 app? The playground in this repo using the above snippet reports a number during runtime (Firefox 116.0.2 and Chromium 115.0.5790.170 on Arch).

![image](https://github.com/vue-youtube/vue-youtube/assets/16667167/57bc3329-f806-4b54-8bbd-d3aac614a4d1)
> Firefox

---

![image](https://github.com/vue-youtube/vue-youtube/assets/16667167/58f03b10-ccd2-4ef2-9134-e54d08b52728)
> Chrome

**Comment by suterma** [comment]
I did now fork the repo and build it for myself. I had a hard time installing pnpm on my Ubuntu box, so I ended up using npm, but I could actually spin up an app instance using the sandbox command. I could not reproduce my error, meaning the reported data in this case has the expected number type.

I can not actually see a relevant difference between my app and the sandbox app, except maybe that I do not use vite yet, but build my stuff using webpack. Maybe there is some subtle difference under the hood somewhere. Maybe it's also a typescript issue, as I am using an older typescript version (it should not, imho).

But it does not really matter to me, and I am not motivated enough to investigate further on my part about this, since I have a perfectly working solution. And it really seems now to not be a general issue. Thanks for your effort.

I am closing my ticket now.
